modules/core/include/opencv2/core/types.hpp

```diff
@@ -886,20 +889,65 @@ class CV_EXPORTS TermCriteria
 
     //! default constructor
     TermCriteria();
+
     /**
-    @param type The type of termination criteria, one of TermCriteria::Type
-    @param maxCount The maximum number of iterations or elements to compute.
-    @param epsilon The desired accuracy or change in parameters at which the iterative algorithm stops.
-    */
+     * @deprecated
+     * @param type The type of termination criteria, one of TermCriteria::Type
+     * @param maxCount The maximum number of iterations or elements to compute.
+     * @param epsilon The desired accuracy or change in parameters at which the iterative algorithm
+     * stops.
+     */
     TermCriteria(int type, int maxCount, double epsilon);
 
+    /**
+     * @param count
+     * @param _epsilon
+     */
+    TermCriteria(int count, double _epsilon);
+
+    /**
+     * Check if flag with according value is set
+     * @deprecated
+     */
     inline bool isValid() const
     {
         const bool isCount = (type & COUNT) && maxCount > 0;
         const bool isEps = (type & EPS) && !cvIsNaN(epsilon);
         return isCount || isEps;
     }
 
+    /**
+     * Convert TermCriteria object from deprecated to actual one
+     * @return isValid() result before conversion
+     */
+    bool fromDeprecated();
+
+    bool checkEpsilonTolerance(double _epsilon) const;
+    bool checkIterationTolerance(int count) const;
+
+    void setEpsilonTolerance(double _epsilon);
```

Comment:
can we come up with some shorter names? `haveEps()`? `haveMaxIters()`? BTW, why not just set maxIters to INT_MAX when it's not set? why not just set eps to 0 if it's not set. Do we have any concrete example of an algorithm where such sophisticated and complex TermCriteria API is essential?

Reply:
> can we come up with some shorter names? 

Just code style question. If shorter names is prefferd in opencv will change it. No problem
> why not just set eps to 0 if it's not set

If eps is not set means that there is no check in eps. And user should know about it
> Do we have any concrete example of an algorithm where such sophisticated and complex TermCriteria API is essential?

I believe that check implementation should be hidden from the user. Also user should be able to easy turn off/on eps/max_iter check without care about thresholds and flags checking every time.
It's just my suggestion